src/Raven.Client/Documents/Session/DocumentSessionRevisions.cs

```diff
@@ -19,6 +21,15 @@ public DocumentSessionRevisions(InMemoryDocumentSessionOperations session) : bas
         {
         }
 
+        public T GetBefore<T>(string id, DateTime before)
```

Comment:
@ayende maybe it would be better to add paging and give user more flexibility here?

```
public List<T> GetBefore<T>(string id, DateTime before, start = 0, int pageSize = 25)
```

Reply:
I specifically wanted to give a single revision here, because multiple revisions are complex.
They are complex because we cannot assume that revisions are sorted by last modified, and I want to avoid creating a new storage index for this. So we do a single search for the latest revisions before the max date and return just that.